app/src/services/queueSvc.js

```diff
@@ -18,129 +18,165 @@
     this.dataService = new DataService();
     this.emailService = new EmailService();
   }
 
-  /** @function connection
-   *  Gets the current QueueConnection
+  /**
+   * @function connection
+   * Gets the current QueueConnection
    */
-  get connection () {
+  get connection() {
     return this._connection;
   }
 
-  /** @function connection
-   *  Sets the current QueueConnection
-   *  also sets the internal queue
-   *  @param {object} v - the QueueConnection
+  /**
+   * @function connection
+   * Sets the current QueueConnection
+   * also sets the internal queue
+   * @param {object} v - the QueueConnection
    */
-  set connection (v) {
+  set connection(v) {
     this._connection = v;
     this._queue = this._connection.queue;
   }
 
-  /** @function dataService
-   *  Gets the current DataService
+  /**
+   * @function dataService
+   * Gets the current DataService
    */
-  get dataService () {
+  get dataService() {
     return this._dataService;
   }
 
-  /** @function dataService
-   *  Sets the current DataService
-   *  @param {object} v - the DataService
+  /**
+   * @function dataService
+   * Sets the current DataService
+   * @param {object} v - the DataService
    */
-  set dataService (v) {
+  set dataService(v) {
     this._dataService = v;
   }
 
-  /** @function emailService
-   *  Gets the current DataService
+  /**
+   * @function emailService
+   * Gets the current DataService
    */
-  get emailService () {
+  get emailService() {
     return this._emailService;
   }
 
-  /** @function emailService
-   *  Sets the current DataService
-   *  @param {object} v - the EmailService
+  /**
+   * @function emailService
+   * Sets the current DataService
+   * @param {object} v - the EmailService
    */
-  set emailService (v) {
+  set emailService(v) {
     this._emailService = v;
   }
 
-  /** @function queue
-   *  Gets the current QueueConnection's queue
+  /**
+   * @function queue
+   * Gets the current QueueConnection's queue
    */
-  get queue () {
+  get queue() {
     return this._queue;
   }
 
-  /** @function enqueue
-   *  Adds a new job to the queue.
-   *  Job contains enough information for the service to read and update data required for the job.
+  /**
+   * @function enqueue
+   * Adds a new job to the queue.
+   * Job contains enough information for the service to read and update data required for the job.
    *
-   *  @param {string} client - the client that owns the message
-   *  @param {object} message - a Message object
-   *  @param {object} opts - Bull opts, including setting delay time
+   * @param {string} client - the client that owns the message
+   * @param {object} message - a Message object
+   * @param {object} opts - Bull opts, including setting delay time
    */
-  async enqueue (client, message, opts = {}) {
-    const job = this.queue.add({
+  async enqueue(client, message, opts = {}) {
+    await this.dataService.updateStatus(client, message.messageId, queueState.ENQUEUED);
```

Comment:
I think we should be fine as is for now - we're currently always calling and awaiting createTransaction before this can be invoked anyways, so catching a NotFoundError here doesn't really accomplish much IMHO as it'd just be reporting on a message not existing. The way it's being used right now it's better to leave as is as it'd allow deeper errors to be passed through and reported in the event something bad does happen.

Reply:
Not sure how well throwing that exception will work in a merge/batch scenario.  The chances of it happening are pretty much nil, but it is still possible.  And looking at it more closely, we should maybe refactor a bit... we should pass along the outside transaction as an optional parameter for the dataService updateStatus.  These enqueue updates should be part of the larger create transaction and messages done in sendEmail/sendEmailMerge

Reply:
I can take another look at this on the next PR as we'll have to encounter the same pattern for handling multiple messages to cancel there too.